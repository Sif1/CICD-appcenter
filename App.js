/* eslint-disable no-alert */

import React from 'react';
import Crashes from 'appcenter-crashes';
import Analytics from 'appcenter-analytics';
import {Button, View, StyleSheet, Text, TextInput} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      inflationRate: 0.0,
      riskFreeRate: 0.0,
      amount: 0.0,
      timeInYears: 1,
      afterInflation: 0.0,
      atRiskFree: 0.0,
      atRiskFreeAfterInflation: 0.0,
      difference: 0,
    };
  }

  componentDidMount() {
    this.checkPreviousSession();
  }

  checkPreviousSession = async () => {
	// useless commenta
    //check if the app has crashed in the last active session
    const didCrash = await Crashes.hasCrashedInLastSession();
    if (didCrash) {
      alert('Sorry about that last crash mate!');
    }
  };

  render() {
    return (
      <View style={styles.container}>
        <Button
          title="Track Event Test"
          onPress={() => Analytics.trackEvent('Test event!')}
        />
        <Text style={styles.text}>HELLO WORLD</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: 40,
    marginHorizontal: 16,
    backgroundColor: '#34e3b7',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 40,
    color: 'black',
  },
});
