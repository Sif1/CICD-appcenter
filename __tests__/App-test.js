/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('appcenter-crashes', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    inFocusDisplaying: jest.fn(),
  };
});

it('renders correctly', () => {
  console.log('------------ UNIT TESTS ---------------');
  //const component = renderer.create(<App />);
  //expect(component).toMatchSnapshot();
  // UNCOMMENT IN ORDER TO FAIL THE TEST/BUILD
});
