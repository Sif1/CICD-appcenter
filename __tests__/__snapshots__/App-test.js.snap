// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 16,
      "marginTop": 40,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Track Event Test
      </Text>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    keyboardType="decimal-pad"
    onChangeText={[Function]}
    placeholder="Current inflation rate"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 30,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="decimal-pad"
    onChangeText={[Function]}
    placeholder="Current risk free rate"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 30,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="decimal-pad"
    onChangeText={[Function]}
    placeholder="Amount you want to save"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 30,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="decimal-pad"
    onChangeText={[Function]}
    placeholder="For how long (in years) will you save?"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 30,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Calculate inflation
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    1
     years from now you will still have $
    0
     but it will only be worth $
    0
    .
  </Text>
  <Text
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    But if you invest it at a risk free rate you will have $
    0
    .
  </Text>
  <Text
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    Which will be worth $
    0
     after inflation.
  </Text>
  <Text
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    A difference of: $
    0
    .
  </Text>
</View>
`;
